service/frontend/admin_handler.go

```diff
@@ -2033,3 +2034,72 @@ func (adh *AdminHandler) StreamWorkflowReplicationMessages(
 	<-shutdownChan.Channel()
 	return nil
 }
+
+func (adh *AdminHandler) GetNamespace(ctx context.Context, request *adminservice.GetNamespaceRequest) (*adminservice.GetNamespaceResponse, error) {
+	req := &persistence.GetNamespaceRequest{
+		Name: request.GetNamespace(),
+		ID:   request.GetId(),
+	}
+	resp, err := adh.persistenceMetadataManager.GetNamespace(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	info := resp.Namespace.GetInfo()
+	nsConfig := resp.Namespace.GetConfig()
+	replicationConfig := resp.Namespace.GetReplicationConfig()
+
+	task := &replicationspb.NamespaceTaskAttributes{
```

Comment:
i believe you should return the namespace as is, then do the conversion after receiving the namespace, i.e. put this conversion logic in the history replication package

Reply:
You are right. Changed the in the new revision.